test/half.h

```diff
@@ -114,7 +114,7 @@ struct half_t
                 { /* normal */
                     ir |= ia >> (24 - 11);
                     ia = ia << (32 - (24 - 11));
-                    ir = ir + ((14 + shift) << 10);
+                    ir = static_cast<uint16_t>(ir + ((14 + shift) << 10));
```

Comment:
This is probably okay because it's just in the test code, but I'm pretty sure this is actually a bug due to potential overflow.